### ISSUE TITLE
Explore random missing issue

### ISSUE THREAD

**Issue by santiagohermo** [diegogentilepassaro/min_wage_rent#73]
A while ago we noticed that some zipcodes have some missing values even after entering the data. We want to explore this issue by

- getting a sense on how common this is.
- studying whether those missing values are random or not.

**Comment by santiagohermo** [comment]
Good news here @diegogentilepassaro @gabrieleborg! I did a brief exploration by creating `enter_panel`, `in_panel` and `missing_in_panel` indicators, and it looks like there are no missing in panel. I'm not sure why I was seeing those a few months ago, but if that ever was a problem it's not in our current estimating panel.

I won't commit the branch here, since I'm not sure we want to add anything. I add the code at the bottom, and show a few results to discuss today. Please let me know if you see anything worth adding to the repo here! Otherwise, I'll close this issue this after our call

- There are no missing in panel

![image](https://user-images.githubusercontent.com/45404755/101026925-7c56a200-3545-11eb-8c67-2681855f3439.png)

- 1,039 MW events happen when our SFCC rent variable is missing

![image](https://user-images.githubusercontent.com/45404755/101027107-b58f1200-3545-11eb-8469-61456ea1c265.png)

- 39 zipcodes enter the panel exactly on the same month as a MW event (0.7% of total events)

![image](https://user-images.githubusercontent.com/45404755/101027486-0d2d7d80-3546-11eb-9ecd-a11822e9ce57.png)

- Running our main model on an `enter_panel` indicator (equal to 1 when the zipcode enters and 0 otherwise), it seems that MW is correlated negatively with entry to the panel. After a MW increase, the probability of entering to the panel drops. Interestingly, there is a weird F5 positive coefficient, similar to the F5 weird negative coefficient when we run the  model on rents. Could they be connected?

![image](https://user-images.githubusercontent.com/45404755/101027818-89c05c00-3546-11eb-98a8-83655d8f7f41.png)


<details><summary>CODE</summary>
<p>

Add the following lines of code in the `preclean.do` program of `first_differences_nlist`

```stata

local rent_var "medrentpricepsqft_sfcc"
gen indicator = (missing(`rent_var') & !missing(L.`rent_var')) | ///
		(!missing(`rent_var') & missing(L.`rent_var')) | ///
		(zipcode != zipcode[_n-1]) // Indicates change of non-missing to missing of viceversa
gen miss_group_id = sum(indicator)  // Creates missing groups ID
bysort zipcode (year_month): egen first_miss_group = min(miss_group_id)  // ID of first missing group by zipcode

gen in_panel = (miss_group_id != first_miss_group )
gen miss_in_panel = (in_panel & missing(`rent_var'))
replace miss_in_panel = . if !in_panel

drop indicator miss_group_id first_missing_group

gen enter_panel = (in_panel == 1 & in_panel[_n-1]==0)
gen mw_event = dactual_mw > 0
```

</p>
</details>

**Comment by santiagohermo** [comment]
Per call, I will try the following:

- Check how STATA is counting missing in the first period
- Make figure with calendar months and share of zipcodes entering on each
- Run dynamic model using data after July 2015 (perfectly balanced panel)
- Estimate model with events not in January and July
- If time permits, run model with changing balancing dates and see how effect changes

More generally, think if we can somehow control for "the clock" that determines MW changes and entry of zipcodes

**Comment by santiagohermo** [comment]
I will finish a pass on the results section tomorrow morning @diegogentilepassaro. If you have bandwidth before then you can try some of the things we discussed in our meeting in this issue. The branch is [`#73-random-missing`](https://github.com/diegogentilepassaro/min_wage_rent/tree/%2373-random-missing)

**Comment by diegogentilepassaro** [comment]
Sure @santiagohermo! I will assign this task to myself now and will report what I've tried :D

**Comment by diegogentilepassaro** [comment]
@santiagohermo  and @gabrieleborg I had some time and was able to do some trials! I committed to a branch some useful code to continue exploration. 

First, I noted that as of January 2010 we have no zipcodes. The first observed zipcodes are in February 2010 and they are only 6. 

Second I plotted the share of zipcodes that enter the sample on each calendar month. As we suspected January is the main month for that, but there is also a mid-year spike. 
![Screen Shot 2020-12-04 at 9 23 45 PM](https://user-images.githubusercontent.com/20582494/101227550-ffc3e080-3676-11eb-8690-f25669045623.png)

Third, I did a table that estimates the baseline dynamic model (without controls) for the sample using zipcode-months from July 2015 and on (a fully balanced panel), and another one using only zipcodes that entered the sample early (during 2010 or 2011). This latter panel is almost fully balanced. I created a table but it is a little rough (it is committed):

![Screen Shot 2020-12-04 at 9 32 04 PM](https://user-images.githubusercontent.com/20582494/101227856-29c9d280-3678-11eb-9acc-354a64282c96.png)

Results are on the same ballpark and that's great news!! We can add a phrase maybe? We still have some experiments on the list though!

**Comment by santiagohermo** [comment]
Thanks @diegogentilepassaro ! That's a very useful exploration (too bad we can't spend too much time on this). So our panel has almost no data in 2010, right? Maybe we can rephrase a bit the paper and say we construct a panel from 2011 to 2020.

On the estimation. What's up with that high F2 coefficient in the second column? It seems that most of our effect is coming from late entrants, right? This makes sense if zipcodes that enter early are also richer and stuff

**Comment by diegogentilepassaro** [comment]
Sorry, I forgot to answer you here @santiagohermo! Let me answer your questions and add a few figures and comments. 

For computing the above figure and table I use our main panel (that cuts zipcodes that entered after 2015m7). Therefore, entrance into the panel is only up to that time. The zipcodes in that main panel are 1143, and of those 110 entered in 2010 (starting in February), and around 704 enter during 2011. After that few zipcodes enter the sample, although for checking how many enter after July 2015 we need to check in the unbalanced panel (if you think it is worth it, I will take a look as well!). Below a year-month and year figure for the number of entrances:
![Screen Shot 2020-12-10 at 1 17 06 PM](https://user-images.githubusercontent.com/20582494/101798633-112f4180-3aea-11eb-89f8-544b28b7cc2f.png)
![Screen Shot 2020-12-10 at 1 17 16 PM](https://user-images.githubusercontent.com/20582494/101798639-12606e80-3aea-11eb-9324-012cd440d756.png)

On the estimation for early entrants only, I am not sure what is going on and we could explore that later! But the 2015m7 and on the fully balanced panel is pretty good because it means that the correlation between MW and entrance is not driving our effects on rents! I created a new table (living in this branch) that we could use to justify a phrase like that where we talk about balancing and entrance in the results section. The table could live in the appendix or not even there, what do you think? Another question I have is if we should add the economic controls once #75 is through?

![Screen Shot 2020-12-10 at 1 33 59 PM](https://user-images.githubusercontent.com/20582494/101800919-9582c400-3aec-11eb-9e13-23cdd87745dc.png)

If you agree let's do a super quick PR and add that phrase somewhere!

**Comment by santiagohermo** [comment]
That's really cool @diegogentilepassaro! It's quite nice to understand when our zipcodes are entering (it seems that Zillow grew a lot in 2011).

The table you posted last is estimated using the fully balanced sample after 2015, right? It's great that our results hold. We definitely should mention this in the paper. Not sure if adding the full table is a good idea since we have many already, but maybe we can think of adding it to the unbalanced/reweighted table at some point (it seems to me that the long-run effect will be larger though)

**Comment by diegogentilepassaro** [comment]
Indeed @santiagohermo ! The table I posted uses the fully balanced panel starting in 2015m7. I agree with you that the table is not necessary, but we should mention it in the paper. Whenever you or @gabrieleborg give me the heads up I'll merge and add a phrase in the balance section of the draft. :D

**Comment by santiagohermo** [comment]
Thanks for the great work here @diegogentilepassaro! I took a quick look at the code and it makes sense to me. I'd say you go ahead and merge to branch #65 here.

Your labels look weird cause the program [make_results_labels](https://github.com/diegogentilepassaro/min_wage_rent/blob/4d9d0cecc58170549f235b030743675d708bf02f/lib/stata/min_wage/ado/make_results_labels.ado) is off-dated. Once you update the branch before merging, re-run and they should fix.

**Comment by gabrieleborg** [comment]
Sorry for the delay @diegogentilepassaro ! 

I also took a look and I agree, it seems fine to me too :) Green light to merge!

**Comment by diegogentilepassaro** [comment]
Merged to #65 in #76
